daemon/create.go

```diff
@@ -19,6 +19,7 @@ func (daemon *Daemon) ContainerCreate(name string, config *runconfig.Config, hos
 		return "", nil, fmt.Errorf("Config cannot be empty in order to create a container")
 	}
 
+	daemon.adaptContainerSettings(hostConfig)
```

Comment:
Ah yes, i think you are correct. The current code will show the warning even if user doesn't specify `MemorySwap` (i.e. MemorySwap == 0 ). Preserving this for now is ok for me.